src/main.cpp

```diff
@@ -3447,6 +3451,23 @@ bool static DisconnectTip(CValidationState& state, const CChainParams& chainpara
         }
     }
 
+    if(pblocktree->GetProposalIndex(vecProposal)){
+
+         for(unsigned int i = 0; i < vecProposal.size(); i++) {
+             propsal = vecProposal[i];
+
+             if(propsal.blockhash == pindexDelete->GetBlockHash()) {
+                 vPropsalsToUpdate.push_back(make_pair(propsal.hash, propsal));
```

Comment:
this line does not make any change to the proposal db, as the same object without modification is inserted with the proposal hash as a key